### PR TITLE
Players in creative and spectator aren't affected by cacti

### LINKED ISSUE #2916: Cannot move well when touching a cactus.
When you run into a cactus you are slowed by more than you should be. Along with being slowed you cannot jump. This also affects people in creative mode even though the movement speed slowing should only affect those in survival or hardcore.

### PR DESCRIPTION
**Pull request by Bond-009**
Fixes #2916

### CODE REVIEW

File: src/Entities/Entity.cpp [hunk 1]
```diff
@@ -1357,6 +1357,14 @@ void cEntity::TickInVoid(cChunk & a_Chunk)
 
 void cEntity::DetectCacti(void)
 {
+	if (
+		IsPlayer() && 
+		((reinterpret_cast<cPlayer *>(this))->IsGameModeCreative() || (reinterpret_cast<cPlayer *>(this))->IsGameModeSpectator()))
```
Comment (Seadragon91) [inline src/Entities/Entity.cpp]: Please change to:
```
if (
	IsPlayer() &&
	((reinterpret_cast<cPlayer *>(this))->IsGameModeCreative() || (reinterpret_cast<cPlayer *>(this))->IsGameModeSpectator())
)
```
The closing round  bracket on a new line.

**Comment by Bond-009** [comment]
What's wrong with CircleCI?

**Comment by Seadragon91** [comment]
No idea...

**Comment by Seadragon91** [comment]
@Bond-009 Please fix the trailing white space. Hope now that circle builds, changed a setting.

**Comment by Seadragon91** [comment]
Okay still doesn't work :(

**Comment by Bond-009** [comment]
Maybe I should move the check to this line.
https://github.com/Bond-009/cuberite/blob/857ca0fa4ae5eab94547a41193b32585c093e7b8/src/Entities/Entity.cpp#L914

**Comment by Seadragon91** [comment]
Thanks for moving the check :)

**Pull request merged**